source/Ship.cpp

```diff
@@ -275,29 +275,61 @@ void Ship::Load(const DataNode &node)
 				if(child.Size() >= 2)
 					outfit = GameData::Outfits().Get(child.Token(1));
 			}
-			Angle gunPortAngle = Angle(0.);
-			bool gunPortParallel = false;
+			Hardpoint::BaseAttributes attributes;
+			attributes.baseAngle = Angle(0.);
+			attributes.isParallel = false;
+			attributes.isOmnidirectional = true;
 			bool drawUnder = (key == "gun");
 			if(child.HasChildren())
 			{
+				bool defaultBaseAngle = true;
 				for(const DataNode &grand : child)
 				{
+					bool needToCheckAngles = false;
 					if(grand.Token(0) == "angle" && grand.Size() >= 2)
-						gunPortAngle = grand.Value(1);
+					{
+						attributes.baseAngle = grand.Value(1);
+						needToCheckAngles = true;
+						defaultBaseAngle = false;
+					}
 					else if(grand.Token(0) == "parallel")
-						gunPortParallel = true;
+						attributes.isParallel = true;
+					else if(grand.Token(0) == "arc" && grand.Size() >= 3)
+					{
+						attributes.isOmnidirectional = false;
+						attributes.minArc = Angle(grand.Value(1));
+						attributes.maxArc = Angle(grand.Value(2));
+						needToCheckAngles = true;
+						if(attributes.minArc.Degrees() > attributes.maxArc.Degrees())
```

Comment:
```suggestion
						if(!Angle(0).IsInRange(attributes.minArc, attributes.maxArc))
```
or
```suggestion
						if(grand.Value(1) > 0.0 || grand.Value(2) < 0.0 || grand.Value(2) - grand.Value(1) >= 360.0)
```
The former allows wrap-around, such as:
```
	turret x y outfit_name
		angle 0
		arc 270 90
```